broker/src/main/java/io/camunda/zeebe/broker/partitioning/PartitionManagerImpl.java

```diff
@@ -188,7 +188,20 @@ private void completePartitionStart(
       if (error instanceof StartupProcessShutdownException) {
         LOGGER.warn("Aborting startup of partition {}", partitionId, error);
       } else {
-        LOGGER.error("Failed to start partition {}", partitionId, error);
+        LOGGER.error(
+            "Failed to start partition {}, removing partition and shutting down already started steps",
+            partitionId,
+            error);
+        concurrencyControl.runOnCompletion(
```

Comment:
:question: This is also applied for normal bootstrapped partitions. But I guess that is ok, because there is no way to recover from it other than restarting the broker?

Reply:
Yeah at first I wasn't sure about this but I think it's fine because the partition will still show up as DEAD and the only recovery option is to restart, as you said.